OpenRA.Mods.Common/Traits/ChangesTerrain.cs

```diff
@@ -0,0 +1,41 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Modifies the terrain type underneath the actors location.")]
+	class ChangesTerrainInfo : ITraitInfo
+	{
+		[FieldLoader.Require] public readonly string TerrainType = null;
+
+		public object Create(ActorInitializer init) { return new ChangesTerrain(this); }
+	}
+
+	class ChangesTerrain : INotifyAddedToWorld
+	{
+		readonly ChangesTerrainInfo info;
+
+		public ChangesTerrain(ChangesTerrainInfo info)
+		{
+			this.info = info;
+		}
+
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
```

Comment:
This is a general purpose trait that will be available to all modders, so i'm not really happy with the idea of merging this with bugs.